frontend/src/lib/k8s/cluster.ts

```diff
@@ -157,18 +165,56 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
 
     static useApiList<U extends KubeObject>(
       onList: (...arg: any[]) => any,
-      onError?: (err: ApiError) => void
+      onError?: (err: ApiError) => void,
+      opts?: ApiListOptions
     ) {
+      const [objs, setObjs] = React.useState<{ [key: string]: U[] }>({});
       const listCallback = onList as (arg: U[]) => void;
-      useConnectApi(this.apiList(listCallback, onError));
+
+      function onObjs(namespace: string, objList: U[]) {
+        if (objList.length > 0) {
+          let newObjs: typeof objs = {};
+          // Set the objects so we have them for the next API response...
+          setObjs(objs => {
+            newObjs = { ...objs, [namespace || '']: objList };
+            return newObjs;
+          });
+
+          let allObjs: U[] = [];
+          Object.values(newObjs).map(currentObjs => {
+            allObjs = allObjs.concat(currentObjs);
+          });
+
+          listCallback(allObjs);
+        }
+      }
+
+      const listCalls = [];
+      if (!!opts?.namespace) {
+        let namespaces: string[] = [];
+        if (typeof opts.namespace === 'string') {
+          namespaces = [opts.namespace];
+        } else {
+          namespaces = opts.namespace as string[];
+        }
+
+        for (const namespace of namespaces) {
+          listCalls.push(
+            this.apiList(objList => onObjs(namespace, objList as U[]), onError, { namespace })
```

Comment:
I wonder what happens with multiple errors? Before onError could be called once, now it can be called multiple times. Not sure it matters.

Reply:
Yeah. This needs some more thought but I think we can proceed with what we have and look into this later. Thanks.